### PR TITLE
Use Twitter REST client

### PR DESCRIPTION
**Pull request by petewest**
I couldn't get this working with the specified version of the Twitter gem as it no longer includes `Twitter.configure` so I've switched it to use `Twitter::REST::Client`.  I (think) I've preserved the original behaviour but I haven't been able to find an example twitter.json so I'm just guessing.

I had to remove the Timeout dependency or it conflicted with Net.

### CODE REVIEW

File: sensu-plugins-twitter.gemspec [hunk 1]
```diff
@@ -39,7 +39,6 @@ Gem::Specification.new do |s|
 
   s.add_runtime_dependency 'sensu-plugin', '1.2.0'
   s.add_runtime_dependency 'twitter',      '5.14.0'
```
Comment (analytically) [inline sensu-plugins-twitter.gemspec]: Mind bumping to 5.15.0 and test?

**Comment by analytically** [comment]
Is this backwards compatible?

File: sensu-plugins-twitter.gemspec (hunk 1 above)
Comment (petewest) [inline sensu-plugins-twitter.gemspec]: Twitter gem v5.15.0 works OK here so I've added a new commit

**Comment by petewest** [comment]
I've only got sensu 0.21.0 to test with so I cannot test backwards compatibility.

**Comment by analytically** [comment]
Is it backward compatible with previous plugin versions and config?

**Comment by petewest** [comment]
It should be, but I can't find any examples of previous configs and couldn't get the previous version working at all so I can't be sure.

**Comment by analytically** [comment]
Cool, thanks, looks good, merging.

**Comment by analytically** [comment]
PS mind sharing your twitter config (fake passwords etc)?

**Comment by petewest** [comment]
Here's my chef template for twitter.json:

``` JSON
{
    "twitter": {
        "<%= @username %>": {
            "consumer_key":  "<%= @consumer_key %>",
            "consumer_secret": "<%= @consumer_secret %>",
            "oauth_token": "<%= @access_token %>",
            "oauth_token_secret": "<%= @access_token_secret %>",
            "sensusub": "twitter"
        }
    }
}
```

"Sensusub" ties in to the client subscription list. I'm guessing so that you can specify different accounts to post different notifications on a per-client basis.

My client config.json has:

``` JSON
{
    "client": {
        "subscriptions": [
            "twitter"
        ]
    }
}
```

**Comment by petewest** [comment]
If you do know anyone that's using the existing plugin it'd be great to see if it works for them.

**Pull request merged**